File: app/src/main/java/com/dialogapp/dialog/util/CustomQuoteSpan.java
package com.dialogapp.dialog.util;

import android.graphics.Canvas;
import android.graphics.Paint;
import android.support.annotation.ColorInt;
import android.text.Layout;
import android.text.style.LeadingMarginSpan;

public class CustomQuoteSpan implements LeadingMarginSpan {
    private int mStripWidth = 2;
    private int mGapWidth = 2;
    private final int mColor;

    public CustomQuoteSpan(@ColorInt int color, int stripeWidth, int gapWidth) {
        super();
        mColor = color;
        mStripWidth = stripeWidth;
        mGapWidth = gapWidth;
    }

    @Override
    public int getLeadingMargin(boolean first) {
        return mStripWidth + mGapWidth;
    }

    @Override
    public void drawLeadingMargin(Canvas c, Paint p, int x, int dir,
                                  int top, int baseline, int bottom,
                                  CharSequence text, int start, int end,
                                  boolean first, Layout layout) {
        Paint.Style style = p.getStyle();
        int color = p.getColor();
        p.setStyle(Paint.Style.FILL);
        p.setColor(mColor);
        c.drawRect(x, top, x + dir * mStripWidth, bottom, p);
        p.setStyle(style);
        p.setColor(color);
    }

    @ColorInt
    public int getColor() {
        return mColor;
    }
}

File: app/src/main/java/com/dialogapp/dialog/ui/util/CustomQuoteSpan.kt
package com.dialogapp.dialog.ui.util

import android.graphics.Canvas
import android.graphics.Paint
import android.text.Layout
import android.text.style.LeadingMarginSpan
import androidx.annotation.ColorInt

class CustomQuoteSpan(@param:ColorInt @get:ColorInt
                      val color: Int, stripeWidth: Int, gapWidth: Int) : LeadingMarginSpan {
    private var mStripWidth = 2
    private var mGapWidth = 2

    init {
        mStripWidth = stripeWidth
        mGapWidth = gapWidth
    }

    override fun getLeadingMargin(first: Boolean): Int {
        return mStripWidth + mGapWidth
    }

    override fun drawLeadingMargin(c: Canvas, p: Paint, x: Int, dir: Int,
                                   top: Int, baseline: Int, bottom: Int,
                                   text: CharSequence, start: Int, end: Int,
                                   first: Boolean, layout: Layout) {
        val style = p.style
        val color = p.color
        p.style = Paint.Style.FILL
        p.color = this.color
        c.drawRect(x.toFloat(), top.toFloat(), (x + dir * mStripWidth).toFloat(), bottom.toFloat(), p)
        p.style = style
        p.color = color
    }
}
